pandas/io/pytables.py

```diff
@@ -1924,6 +1926,13 @@ def get_atom_coltype(self, kind=None):
         else:
             col_name = "%sCol" % kind.capitalize()
 
+        if kind.startswith('complex'):
```

Comment:
I would do this in ``set_atom``, ``block.is_complex`` is by definition already true at that point.